structures/aggregatecontents.go

```diff
@@ -0,0 +1,163 @@
+		{0, 0, 0, 4, 10, 22, 27, 33, 44, 48}, //non-perishable
+		{0, 0, 0, 5, 17, 28, 37, 43, 50, 50}} //perishable
+	min := transpose(contentDamageFunctionArrayMin)
+	// 2D array of maximum content damage engineered structures
+	contentDamageFunctionArrayMax := [][]float64{
+		{0, 0, 5, 15, 22, 35, 44, 50, 55, 70}, //non-perishable
+		{0, 0, 8, 28, 50, 58, 65, 65, 90, 90}} //perishable
+	max := transpose(contentDamageFunctionArrayMax)
+
+	structurexs := []float64{-1, -0.5, 0, 0.5, 1, 2, 3, 5, 7, 10}
+	structureydists := make([]statistics.ContinuousDistribution, 10)
+	structureydists[0] = statistics.TriangularDistribution{Min: 0, MostLikely: 0, Max: 0}
+	structureydists[1] = statistics.TriangularDistribution{Min: 0, MostLikely: 0, Max: 0}
+	structureydists[2] = statistics.TriangularDistribution{Min: 0, MostLikely: 5, Max: 9}
+	structureydists[3] = statistics.TriangularDistribution{Min: 5, MostLikely: 10, Max: 17}
+	structureydists[4] = statistics.TriangularDistribution{Min: 12, MostLikely: 20, Max: 27}
+	structureydists[5] = statistics.TriangularDistribution{Min: 18, MostLikely: 30, Max: 36}
+	structureydists[6] = statistics.TriangularDistribution{Min: 28, MostLikely: 35, Max: 43}
+	structureydists[7] = statistics.TriangularDistribution{Min: 33, MostLikely: 40, Max: 48}
+	structureydists[8] = statistics.TriangularDistribution{Min: 43, MostLikely: 53, Max: 60}
+	structureydists[9] = statistics.TriangularDistribution{Min: 48, MostLikely: 58, Max: 69}
+	contentxs := []float64{-1, -0.5, 0, 0.5, 1, 2, 3, 5, 7, 10}
+	contentydists := make([]statistics.ContinuousDistribution, 10)
+	for i := 0; i < len(contentxs); i++ {
+		contentydists[i] = aggregateTriangular(min[i], mostLikely[i], max[i])
+	}
+	var structuredamagefunctionStochastic = paireddata.UncertaintyPairedData{Xvals: structurexs, Yvals: structureydists}
+	var contentdamagefunctionStochastic = paireddata.UncertaintyPairedData{Xvals: contentxs, Yvals: contentydists}
+
+	sm := make(map[hazards.Parameter]interface{})
+	var sdf = structures.DamageFunctionFamilyStochastic{DamageFunctions: sm}
+
+	cm := make(map[hazards.Parameter]interface{})
+	var cdf = structures.DamageFunctionFamilyStochastic{DamageFunctions: cm}
+	//Default hazard.
+	sdf.DamageFunctions[hazards.Default] = structuredamagefunctionStochastic
+	cdf.DamageFunctions[hazards.Default] = contentdamagefunctionStochastic
+	//Depth Hazard
+	sdf.DamageFunctions[hazards.Depth] = structuredamagefunctionStochastic
+	cdf.DamageFunctions[hazards.Depth] = contentdamagefunctionStochastic
+
+	return structures.OccupancyTypeStochastic{Name: "comEng", StructureDFF: sdf, ContentDFF: cdf}
+}
+
+func comNonEng() structures.OccupancyTypeStochastic {
+
+	// 2D array of most likely content damage engineeered structures
+	contentDamageFunctionArrayMostLikely := [][]float64{
+		{0, 0, 1, 8, 12, 18, 25, 39, 50, 60},  //non-perishable
+		{0, 0, 2, 15, 30, 42, 64, 71, 80, 87}} //perishable
+	mostLikely := transpose(contentDamageFunctionArrayMostLikely)
```

Comment:
i think we want to do this compute once outside of go_consequences, store the result, analyze the result, and then determine a slice of distributions to use. doing this compute each time we ask for occupancy type maps will slow down our compute considerably.